### PR TITLE
CIV-9040 Redetermination - Interlocutory Judgement PDF (docmosis)

### PR DESCRIPTION
**Pull request by manish14836**
Files to be reviewed 

src/main/common/models/ccdResponse/ccdPayBySetDate.ts
src/main/common/models/events/eventDto.ts
src/main/routes/features/claimantResponse/ccj/judgmentAmountSummaryExtendedController.ts
src/main/routes/features/claimantResponse/claimantSuggestedPaymentDateController.ts
src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts
src/main/services/translation/claimantResponse/convertToCCDClaimantLiPResponse.ts
src/main/services/translation/claimantResponse/convertToCCDClaimantPaymentOption.ts
src/main/services/translation/convertToCUI/cuiTranslation.ts
src/main/services/translation/response/convertToCCDPayBySetDate.ts
src/test/unit/common/models/claim.test.ts
src/test/unit/routes/features/claimantResponse/ccj/judgmentAmountSummaryExtendedController.test.ts
src/test/unit/services/translation/convertToCUI/cuiTranslation.test.ts

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/services/translation/claimantResponse/convertToCCDClaimantLiPResponse.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {ClaimantResponse} from 'common/models/claimantResponse';
 import {YesNoUpperCamelCase} from 'form/models/yesNo';
 import {toCCDYesNoFromBooleanString} from 'services/translation/response/convertToCCDYesNo';
 import {ChooseHowProceed} from 'common/models/chooseHowProceed';
+import { RepaymentDecisionType } from 'common/models/claimantResponse/RepaymentDecisionType';
```
Comment (jarekPierchala) [inline src/main/services/translation/claimantResponse/convertToCCDClaimantLiPResponse.ts:9]: and here

File: src/main/services/translation/claimantResponse/convertToCCDClaimantPaymentOption.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import {PaymentOptionType} from 'common/form/models/admission/paymentOption/paymentOptionType';
+import { ClaimantResponse } from 'common/models/claimantResponse';
```
Comment (jarekPierchala) [inline src/main/services/translation/claimantResponse/convertToCCDClaimantPaymentOption.ts:2]: .....

File: src/main/routes/features/claimantResponse/ccj/judgmentAmountSummaryExtendedController.ts [hunk 3]
```diff
@@ -3,12 +3,14 @@ import {
...
   CLAIMANT_RESPONSE_TASK_LIST_URL,
 } from 'routes/urls';
-import {generateRedisKey, getCaseDataFromStore} from 'modules/draft-store/draftStoreService';
+import {generateRedisKey, getCaseDataFromStore, saveDraftClaim} from 'modules/draft-store/draftStoreService';
 import {AppRequest} from 'models/AppRequest';
 import {Claim} from 'models/claim';
 import {getJudgmentAmountSummary} from 'services/features/claimantResponse/ccj/judgmentAmountSummaryService';
 import {constructResponseUrlWithIdParams} from 'common/utils/urlFormatter';
 import {convertToPoundsFilter} from 'common/utils/currencyFormat';
+import { ChooseHowProceed } from 'common/models/chooseHowProceed';
```
[the remark this answers is not in the document]
Comment (manish14836) [reply src/main/routes/features/claimantResponse/ccj/judgmentAmountSummaryExtendedController.ts]: Removed space

File: src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts [hunk 4]
```diff
@@ -9,7 +9,10 @@ import {toCCDYesNo} from '../response/convertToCCDYesNo';
 import {toCCDExpert} from '../response/convertToCCDExpert';
 import {CCDClaimantResponse} from 'common/models/claimantResponse/ccdClaimantResponse';
 import {toCCDClaimantMediation} from './convertToCCDClaimantMediation';
-import {toCCDClaimantPaymentOption} from 'services/translation/claimantResponse/convertToCCDClaimantPaymentOption';
+import {toCCDClaimantPaymentOption, toCCDClaimantSuggestedPayByDate, toCCDClaimantSuggestedPayByInstalments} from 'services/translation/claimantResponse/convertToCCDClaimantPaymentOption';
+import {toCCDClaimantPayBySetDate} from '../response/convertToCCDPayBySetDate';
+import { toCCDRepaymentPlanFrequency } from '../response/convertToCCDRepaymentPlan';
```
[the remark this answers is not in the document]
Comment (manish14836) [reply src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts]: Removed space

**Review by kenneth-hmcts — APPROVED** [review]

**Review by kenneth-hmcts — DISMISSED** [review]

**Review by ElifDenizEsea — APPROVED** [review]

File: src/main/services/translation/claimantResponse/convertToCCDClaimantLiPResponse.ts [hunk 5]
```diff
@@ -30,5 +33,7 @@ export const toCCDClaimantLiPResponse = (claimantResponse: ClaimantResponse): CC
     applicant1DQHearingSupportLip: toCCDSHearingSupport(claimantResponse?.directionQuestionnaire?.hearing?.supportRequiredList),
     applicant1ChoosesHowToProceed: toChoosesHowToProceed[claimantResponse?.chooseHowToProceed?.option],
     applicant1SignedSettlementAgreement: toCCDYesNoFromBooleanString(claimantResponse?.signSettlementAgreement?.signed),
+    applicant1RejectedRepaymentReason: claimantResponse?.rejectionReason?.text,
+    claimantCourtDecision : claimantResponse?.courtDecision,
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/convertToCCDClaimantLiPResponse.ts:37]: This claimantCourtDecision is added as part of CIV-11733 https://github.com/hmcts/civil-citizen-ui/pull/2672/files

Comment (manish14836) [reply src/main/services/translation/claimantResponse/convertToCCDClaimantLiPResponse.ts:37]: yes, I need this to move in my task so I have included the changes in my PR's.

File: src/main/services/translation/response/convertToCCDPayBySetDate.ts [hunk 6]
```diff
@@ -13,3 +13,9 @@ export const toCCDPayBySetDate = (paymentDate: Date, paymentOption: PaymentOptio
     };
   }
 };
+
+export const toCCDClaimantPayBySetDate = (paymentDate: Date): CCDClaimantPayBySetDate => {
+  return {
+    paymentSetDate: paymentDate,
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/response/convertToCCDPayBySetDate.ts:19]: Generally, in ccd, we will store the date 'yyyy-MM-dd'  in this format with type string, why it is in a different way example "2023-12-07T13:13:40"

Comment (manish14836) [reply src/main/services/translation/response/convertToCCDPayBySetDate.ts:19]: we are sending this field as date to civil service and civil service is storing this in same format to database.

**Review by jeswanth-hmcts — DISMISSED** [review]

**Review by jarekPierchala — DISMISSED** [review]

**Review by jeswanth-hmcts — DISMISSED** [review]

**Review by kalachandrasekar1 — APPROVED** [review]

File: src/main/services/translation/response/convertToCCDPayBySetDate.ts (hunk 6 above)
Comment (manish14836) [reply src/main/services/translation/response/convertToCCDPayBySetDate.ts]: Updated to string type

**Review by neeta-hmcts — APPROVED** [review]

**Review by kalachandrasekar1 — DISMISSED** [review]

**Pull request merged**